src/plugins/walletConnectToDapps/components/DappRegistryGrid.tsx

```diff
@@ -0,0 +1,96 @@
+import { SearchIcon } from '@chakra-ui/icons'
+import {
+  Box,
+  Heading,
+  Image,
+  Input,
+  InputGroup,
+  InputLeftElement,
+  Link,
+  SimpleGrid,
+  Stack,
+  Text as PlainText
+} from '@chakra-ui/react'
+import { Text } from 'components/Text'
+import { FC, useEffect, useMemo } from 'react'
+import { useForm } from 'react-hook-form'
+
+interface RegistryItem {
+  category: string
+  id: string
+  homepage: string
+  name: string
+  image: string
+}
+
+const registry: RegistryItem[] = require('../registry.json')
+
+const pageSize = 20
+
+export const DappRegistryGrid: FC = () => {
+  const { register, watch, setValue } = useForm<{ search: string; page: number }>({
+    mode: 'onChange',
+    defaultValues: {
+      search: '',
+      page: 0,
+    },
+  })
+
+  const search = watch('search')
+  const page = watch('page')
+  useEffect(() => setValue('page', 0), [search, setValue])
+
+  const filteredListings = useMemo(
+    () => registry.filter(l => !search || l.name.toLowerCase().includes(search.toLowerCase())),
```

Comment:
Use full variable names in callbacks 🙏🏽 

```suggestion
    () => registryItems.filter(registryItem => !search || registryItem.name.toLowerCase().includes(search.toLowerCase())),
```